addons/hr_attendance/report/attendance_errors.py

```diff
@@ -3,63 +3,66 @@
-        res = self.cr.dictfetchall()
-        for r in res:
-            if r['action'] == 'sign_out':
-                r['delay'] = -r['delay']
-            temp = r['delay'].seconds
+class ReportHrAttendanceErrors(models.AbstractModel):
+    _name = 'report.hr_attendance.report_attendanceerrors'
+    _inherit = 'report.abstract_report'
+    _template = 'hr_attendance.report_attendanceerrors'
 
-            r['delay'] = str(r['delay']).split('.')[0]
-            if abs(temp) < max*60:
-                r['delay2'] = r['delay']
-            else:
-                r['delay2'] = '/'
+    def _delay(self, employee_id, dt_from, dt_to):
+        res = {}
+        attendances = self.env['hr.attendance'].search([('employee_id', '=', employee_id), ('name', '<=', dt_to), ('name', '>=', dt_from), ('action', 'in', ('sign_in', 'sign_out'))], order="name")
+        for attend in attendances:
+            name = fields.Datetime.from_string(attend.name)
+            create_date = fields.Datetime.from_string(attend.create_date)
+            delay_time = (name - create_date)
+            res.update({'action': attend.action, 'create_date': create_date, 'date': name, 'delay': delay_time})
+            if attend.action == 'sign_out':
+                res['delay'] = -res['delay']
         return res
 
-    def _lst_total(self, employee_id, dt_from, dt_to, max, *args):
-        self.cr.execute("select name as date, create_date, action, create_date-name as delay from hr_attendance where employee_id=%s and to_char(name,'YYYY-mm-dd')<=%s and to_char(name,'YYYY-mm-dd')>=%s and action IN (%s,%s) order by name", (employee_id, dt_to, dt_from, 'sign_in', 'sign_out'))
-        res = self.cr.dictfetchall()
-        if not res:
-            return ('/','/')
-        total2 = datetime.timedelta(seconds = 0, minutes = 0, hours = 0)
-        total = datetime.timedelta(seconds = 0, minutes = 0, hours = 0)
-        for r in res:
-            if r['action'] == 'sign_out':
-                r['delay'] = -r['delay']
-            total += r['delay']
-            if abs(r['delay'].seconds) < max*60:
-                total2 += r['delay']
+    def _lst(self, employee_id, dt_from, dt_to, max_delay):
+        res = self._delay(employee_id, dt_from, dt_to)
+        temp = res['delay'].seconds
+        res['delay'] = str(res['delay']).split('.')[0]
+        if abs(temp) < max_delay*60:
+            res['delay2'] = res['delay']
+        else:
+            res['delay2'] = '/'
+        return [res]
 
-        result_dict = {
-                'total': total and str(total).split('.')[0],
-                'total2': total2  and str(total2).split('.')[0]
-                }
-        return [result_dict]
+    def _lst_total(self, employee_id, dt_from, dt_to, max_delay):
+        total2 = datetime.timedelta(seconds=0, minutes=0, hours=0)
+        total = datetime.timedelta(seconds=0, minutes=0, hours=0)
+        res = self._delay(employee_id, dt_from, dt_to)
```

Comment:
why not iterating over result, _delay may return more then one record(not that _delay does search based on domain and it may return more than one record)